hyperjumptech/monika
# Fix for Bug/330  double logs on  multiple alerts

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Closing #330 
2. Fix for double logs when multiple alerts per probe tiriggered

## How did you implement / how did you fix it  
1. Moved the log reporting to AFTER all the alerts have been checked.
2. Previous implementation has been to print logs after each alerts, so for probes with multiple alerts, a log was printed for each.
3. Also removed redundant log message setting from previous mLog implementations 

## How to test  
1. have a setup with two alert triggers, ie: response. "query": "response.status != 200" and "response.time > 70"
2. run with monika (up to v1.6.0) with repeat = 1 
3. run on this PR with the same config and r=1. You should get the below results.

Before the fix (ran with monika v1.6.0):
![image](https://user-images.githubusercontent.com/964106/132974071-6ddfab74-e8f3-402a-a66e-8ebbb992b55f.png)

after the fix:
![image](https://user-images.githubusercontent.com/964106/132974140-2ddca69a-8159-4a3d-af59-18692991535f.png)
